### PR TITLE
Test for the common functions

### PR DESCRIPTION
**Pull request by AnnaDaiZH**
- added tests for the used common functions 
- deleted unused common functions
- fixed minor bugs in elevationAngletoIndex, azimuthAngletoIndex and velocityLinear
- changed some variable types from int to float, to enable more accurate calculation which also removes rounding error

### CODE REVIEW

File: local_planner/src/nodes/common.cpp [hunk 1]
```diff
@@ -179,31 +156,18 @@ void wrapAngleToPlusMinusPI(double& angle) {
...
 
   double vel;
-  if (std::abs(yaw_vel1) < std::abs(yaw_vel2)) {
-    vel = yaw_vel1;
-  } else {
-    vel = yaw_vel2;
-  }
-
+  // check which yaw direction is shorter
+  std::abs(yaw_vel1) <= std::abs(yaw_vel2)? vel = yaw_vel1: vel = yaw_vel2;
```
Comment (jkflying) [inline local_planner/src/nodes/common.cpp]: This should be
` vel = (std::abs(yaw_vel1) <= std::abs(yaw_vel2))? yaw_vel1 : yaw_vel2;`

File: local_planner/src/nodes/common.cpp [hunk 2]
```diff
@@ -75,46 +51,54 @@ double azimuthIndexToAngle(int z, double res) {
...
+int azimuthAngletoIndex(float z, int res) {  //[-180,180]
+  if(res <= 0.f || z < -180.f || z > 180.f){
+    return 0.f;
+  }
+  if (z == 180.f) {
     z = -180;
   }
   z += 180;
-  z = z + (res - (z % res));  //[-80,+90]
+  z = z + (res - ((int)z % res));  //[-80,+90]
```
Comment (nicovanduijn) [inline local_planner/src/nodes/common.cpp]: c-style casts are not preferred in google style (https://google.github.io/styleguide/cppguide.html#Casting). use `static_cast<int>(z)`

File: local_planner/src/nodes/common.h [hunk 3]
```diff
@@ -90,7 +90,19 @@ void normalize(geometry_msgs::Point& p);
 **/
 double velocityLinear(double max_vel, double slope,
                       double v_old, double elapsed);
+/**
+* @brief     wrappes the input angle in to plus minus PI space
+* @param[in] angle to be wrapped  [rad]
+* @returns   angle wrapped [rad]
```
Comment (nicovanduijn) [inline local_planner/src/nodes/common.h]: this function doesn't return anything. instead write:
`@param[in,out] angle to be wrapped [rad]`

File: local_planner/test/test_common.cpp [hunk 4]
```diff
@@ -26,3 +34,369 @@ TEST(Common, polar2DdistanceOnKnownPoints) {
...
+geometry_msgs::Point pos;
+pos.x = 0.f;
+pos.y = 0.f;
+pos.z = 0.f;
+
+std::vector<geometry_msgs::Point> pos_out;
+
+//WHEN: converting the point in polar CS to cartesian CS
+
+for(int i = 0; i<e.size(); i++){
```
Comment (nicovanduijn) [inline local_planner/test/test_common.cpp]: use a range-based for loop

File: local_planner/test/test_common.cpp [hunk 5]
```diff
@@ -26,3 +34,369 @@ TEST(Common, polar2DdistanceOnKnownPoints) {
...
+     pos_out.push_back(fromPolarToCartesian(e[i], z[3], radius[0], pos));
+}
+
+for(int i = 0; i<e.size(); i++){  
+       pos_out.push_back(fromPolarToCartesian(e[i], z[i], radius[1], pos));
```
Comment (nicovanduijn) [inline local_planner/test/test_common.cpp]: This is dangerous! There's nothing guaranteeing you that `z` has the same dimensions as `e`, so `z[i]` could segfault

File: local_planner/test/test_common.cpp [hunk 6]
```diff
@@ -26,3 +34,369 @@ TEST(Common, polar2DdistanceOnKnownPoints) {
...
+}
+
+
+//THEN: the cartesian coordinates are
+
+for(int i = 0; i<e.size(); i++){
```
Comment (nicovanduijn) [inline local_planner/test/test_common.cpp]: Again, checking for the size of `e` when actually iterating over some other vector is not very safe. Consider something like `for (auto i : pos_out){`

File: local_planner/src/nodes/common.cpp [hunk 7]
```diff
@@ -75,46 +52,54 @@ double azimuthIndexToAngle(int z, double res) {
...
                                 const geometry_msgs::Point& origin) {
   return elevationAnglefromCartesian(pos.x, pos.y, pos.z, origin);
 }
 
-int elevationAngletoIndex(int e, int res) {  //[-90,90]
-  if (e == 90) {
+int elevationAngletoIndex(float e, int res) {  //[-90,90]
+  // Do some input-checks
+  if(res <= 0.f || e < -90.f || e > 90.f){
+    return 0.f;
```
[the remark this answers is not in the document]
Comment (AnnaDaiZH) [reply local_planner/src/nodes/common.cpp]: I changed the function to return a bool. Further i made the adaption of wrapping the elevation angle when input angle larger than 90/-90 but smaller than 180/-180. this needs to be tested first.

File: local_planner/src/nodes/common.cpp [hunk 8]
```diff
@@ -75,46 +52,54 @@ double azimuthIndexToAngle(int z, double res) {
...
-  e = e + (res - (e % res));  //[-80,+90]
-  return e / res - 1;         //[0,17]
+  e = e + (res - ((int)e % res));  //[-80,+90]
+  return floor(e / res )- 1;         //[0,17]
 }
 
-int azimuthAngletoIndex(int z, int res) {  //[-180,180]
-  if (z == 180) {
+int azimuthAngletoIndex(float z, int res) {  //[-180,180]
+  if(res <= 0.f || z < -180.f || z > 180.f){
```
[the remark this answers is not in the document]
Comment (AnnaDaiZH) [reply local_planner/src/nodes/common.cpp]: The problem is you cannot catch the case for invalid resolutions

File: local_planner/src/nodes/local_planner.cpp [hunk 9]
```diff
@@ -263,9 +263,9 @@ void LocalPlanner::determineStrategy() {
         int n_occupied_cells = 0;
 
         int goal_e_angle =
-            elevationAnglefromCartesian(goal_, pose_.pose.position);
+            floor(elevationAnglefromCartesian(goal_, pose_.pose.position));
         int goal_z_angle =
-            azimuthAnglefromCartesian(goal_, pose_.pose.position);
+            floor(azimuthAnglefromCartesian(goal_, pose_.pose.position));
```
[the remark this answers is not in the document]
Comment (AnnaDaiZH) [reply local_planner/src/nodes/local_planner.cpp]: No, changed in new commit

File: local_planner/src/nodes/star_planner.cpp [hunk 10]
```diff
@@ -111,8 +111,8 @@ double StarPlanner::treeCostFunction(int node_number) {
 }
 double StarPlanner::treeHeuristicFunction(int node_number) {
   geometry_msgs::Point node_position = tree_[node_number].getPosition();
-  int goal_z = azimuthAnglefromCartesian(goal_, node_position);
-  int goal_e = elevationAnglefromCartesian(goal_, node_position);
+  int goal_z = floor(azimuthAnglefromCartesian(goal_, node_position));
+  int goal_e = floor(elevationAnglefromCartesian(goal_, node_position));
```
[the remark this answers is not in the document]
Comment (AnnaDaiZH) [reply local_planner/src/nodes/star_planner.cpp]: The results are the same. Removed it in new commit

File: local_planner/src/nodes/common.cpp (hunk 2 above)
Comment (baumanta) [reply local_planner/src/nodes/common.cpp]: Agree, but I think we should leave that for now and open a new PR where we change it consistently in the whole code

File: local_planner/test/test_common.cpp (hunk 5 above)
Comment (AnnaDaiZH) [reply local_planner/test/test_common.cpp]: i have changed it to the range based for loops.  I just haven't pushed it yet.

Comment (AnnaDaiZH) [reply local_planner/test/test_common.cpp]: the change will come in tomorrow

File: local_planner/src/nodes/common.cpp [hunk 11]
```diff
@@ -152,23 +137,15 @@ void normalize(geometry_msgs::Point& p) {
...
 
-double velocityLinear(double max_vel, double min_vel, double slope,
+double velocityLinear(double max_vel,  double slope,
                       double v_old, double elapsed) {
-  v_old -= min_vel;
-  double t_old = v_old / slope;
-  double t_new = t_old + elapsed;
-  double speed = min_vel + t_new * slope;
+  double t_old = v_old / slope; // slope hard-coded as 1.f
+  double t_new = t_old + elapsed; //v_old + elapsed
```
[the remark this answers is not in the document]
Comment (AnnaDaiZH) [reply local_planner/src/nodes/common.cpp]: removed

**Pull request merged**